### DISCUSSION TITLE
umijs 4打包时怎么选用 dynamicImport: {},

### DISCUSSION THREAD

**Discussion by aaron-weldoor** [umijs/umi#8257] [General]
umijs3 在 app.tx 直接用 dynamicImport: {},
请问 umijs4怎样实现此功能？

**Comment by fz6m** [comment]
umi4 默认按页拆包，近似等同于 dynamicImport ，使用 `src/loading.tsx` 自定义加载动画

**Comment by skitsanos** [comment]
That's actually an undocumented feature. In umi3 it was part of the config and you would set it in `.umirc.js` like this:

```json
{
"dynamicImport": {
     "loading": "@/components/Loading"
     }
}
```

I think you need to update https://umijs.org/docs/guides/directory-structure page with details about it as well.
